bot/exts/source.py

```diff
@@ -0,0 +1,100 @@
+import inspect
+from pathlib import Path
+from typing import Optional, Tuple
+
+from discord import Embed
+from discord.ext import commands
+
+from bot.bot import SirRobin
+from bot.constants import URLs
+from bot.converters import SourceConverter, SourceType
+
+
+class BotSource(commands.Cog):
+    """Displays information about the bot's source code."""
+
+    def __init__(self, bot: SirRobin):
+        self.bot = bot
+
+    @commands.command(name="source", aliases=("src",))
+    async def source_command(self, ctx: commands.Context, *, source_item: SourceConverter = None) -> None:
+        """Display information and a GitHub link to the source code of a command, tag, or cog."""
+        if not source_item:
+            embed = Embed(title="Sir Robin's GitHub Repository")
+            embed.add_field(name="Repository", value=f"[Go to GitHub]({URLs.github_bot_repo})")
+            embed.set_thumbnail(url="https://avatars1.githubusercontent.com/u/9919")
+            await ctx.send(embed=embed)
+            return
+
+        embed = await self.build_embed(source_item)
+        await ctx.send(embed=embed)
+
+    def get_source_link(self, source_item: SourceType) -> Tuple[str, str, Optional[int]]:
```

Comment:
str is included twice in the `Tuple` typing?

Reply:
```python
return url, file_location, first_line_no or None
```
That's the return statement of the mentioned function

Reply:
It's correct since this method returns a tuple in which the first and second elements are `str`s.

Reply:
oh lol, my brain was thinking `Union` oopsie. Thanks guys

Reply:
I promise I know what a `Tuple` is 🤣